src/App.tsx

```diff
@@ -1,42 +1,176 @@
+import { NavigationContainer } from "@react-navigation/native";
+import { createNativeStackNavigator } from "@react-navigation/native-stack";
+import {
+  type BottomTabNavigationOptions,
+  createBottomTabNavigator,
+} from "@react-navigation/bottom-tabs";
+import { userProfiles } from "@/services/profile.service";
+import Octicons from "@expo/vector-icons/Octicons";
+import WelcomeScreen from "@/pages/onboarding/welcome/Welcome";
+import ConnectionDetailScreen from "@/pages/default/connections/ConnectionDetail";
+import ReviewConnectionScreen from "@/pages/default/connections/ReviewConnection";
+import AddCredentialsScreen from "@/pages/default/credentials/AddCredentials";
+import CredentialDetailScreen from "@/pages/default/credentials/CredentialDetail";
+import DiscoverScreen from "@/pages/default/discover/Discover";
+import ProfileDetailScreen from "@/pages/default/profiles/ProfileDetail";
+import AuthScreen from "@/pages/onboarding/auth/Auth";
+import CreateScreen from "@/pages/onboarding/create/Create";
+import ImportScreen from "@/pages/onboarding/import/Import";
+import ProfilesScreen from "@/pages/default/profiles/Profiles";
+import CredentialsScreen from "@/pages/default/credentials/Credentials";
+import ConnectionsScreen from "@/pages/default/connections/Connections";
+import ActivityScreen from "@/pages/default/activity/Activity";
+import { type LinkingOptions } from "@react-navigation/native";
+import { ColorTheme, DefaultTheme } from "@/theme/colors";
+import { Space } from "@/theme/layouts";
+import { Typography } from "@/theme/typography";
+import AddProfileScreen from "@/pages/default/profiles/AddProfile";
+import AddCredentialDetailScreen from "./pages/default/credentials/AddCredentialDetail";
+import AddCredentialOptionsScreen from "./pages/default/credentials/AddCredentialOptions";
 
-export default function App() {
-  useEffect(() => {
-    const startupTasks = async () => {
-      await DwnService.initExpoLevelDwn();
-      await bootstrapIdentityAgent(
-        "passphrase",
-        "Personal",
-        DwnService.getDwn()
-      );
-    };
-    void startupTasks();
-  }, []);
+const App = () => {
+  const Stack = createNativeStackNavigator();
+  const initialRoute = userProfiles.peek().length ? "Default" : "Onboarding";
 
   return (
-    <NavigationContainer theme={DarkTheme} linking={linking}>
-      <StatusBar style="light" />
-      <PaperProvider theme={theme}>
-        <AppNavigator />
-      </PaperProvider>
+    <NavigationContainer linking={DeepLinkConfig} theme={DefaultTheme}>
+      <Stack.Navigator
+        screenOptions={{ headerShown: false }}
+        initialRouteName={initialRoute}
+      >
+        {/* Onboarding */}
+        <Stack.Screen name="Onboarding" component={WelcomeScreen} />
+        <Stack.Screen name="Import" component={ImportScreen} />
+        <Stack.Screen name="Create" component={CreateScreen} />
+        <Stack.Screen name="Auth" component={AuthScreen} />
+
+        {/* Default */}
```

Comment:
we can actually split up your groups into files so we don't have to do this for legibility